corehq/apps/app_manager/tasks.py

```diff
@@ -124,21 +123,34 @@ def check_build_dependencies(new_build):
     """
     Reports whether the app dependencies have been added or removed.
     """
+    from corehq.apps.app_manager.models import Application
 
     def has_dependencies(build):
         return bool(
             build.profile.get('features', {}).get('dependencies')
         )
 
-    new_build_has_dependencies = has_dependencies(new_build)
-    app_build_ids = get_build_ids(new_build.domain, new_build.copy_of)
+    def get_previous_build():
+        domain = new_build.domain
+        app_id = new_build.copy_of
+
+        previous_build_doc = Application.get_db().view(
+            'app_manager/saved_app',
+            startkey=[domain, app_id, new_build.id],
+            endkey=[domain, app_id],
+            descending=True,
+            limit=1,
+            skip=1,
+        ).first()
```

Comment:
just confirming that this returns None in case of no previous build? And you tested that I believe for a fresh app then?

Reply:
Additionally, can we further improve this by using 
```
reduce=False,
include_docs=False
```
I believe the key itself would have the ID of the previous build, so you don't need the doc.
Or if works you can wrap the doc returned.

Reply:
> just confirming that this returns None in case of no previous build?

Yes, it returns `None`.

> Additionally, can we further improve this by using

The `include_docs` key has a default of `False`, but not sure about `reduce`. Will update.

Reply:
oh feel free to check what it does, its been a while for me and I remember that these both were to be used together.

Reply:
>  include_docs key has a default of False

Oh cool, does this view then return a key? and not a doc? I was confused by the naming of `previous_build_doc`

Reply:
If `include_docs` is `True` it will return the whole doc with properties etc, otherwise it only returns a very brief version of what looks like metadata.

Reply:
what does it return if include_docs is False? can you run one instance and see what you get to be sure?

>  looks like metadata.

I believe this should be the key it stored.